### PR TITLE
Handler3 pre_/post_action results are not ignored.

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**
Exceptions in actions are handled.

### CODE REVIEW

File: include/thread_pool.h [hunk 1]
```diff
@@ -40,14 +40,30 @@ class GraftJob
...
-            status_ref = h3_ref.worker_action(vars_cref, input_ref, ctx, output_ref);
+            try
+            {
+                ctx.local.last_status = h3_ref.worker_action(vars_cref, input_ref, ctx, output_ref);
+                if(Status::Ok == ctx.local.last_status && h3_ref.post_action || Status::Forward == ctx.local.last_status)
+                {
+                    input_ref.assign(output_ref);
+                }
+            }
+            catch(std::exception& e)
```
Comment (mbg033) [inline include/thread_pool.h]: Catch exception by const reference, as it most likely wouldn't be modified?

File: include/context.h [hunk 2]
```diff
@@ -90,6 +91,14 @@ struct Context
         {
             m_map.erase(key);
         }
+        void set_error(const char* str, Status status = Status::InternalError)
```
Comment (mbg033) [inline include/context.h]: code style: do we use camel case naming or snake case naming (underscores) in public interface ? Guess we should define/select one or another, mixing both doesn't look good

File: include/thread_pool.h [hunk 3]
```diff
@@ -40,14 +40,31 @@ class GraftJob
...
+                }
+            }
+            catch(const std::exception& e)
+            {
+                ctx.local.setError(e.what());
+                input_ref.reset();
+            }
+            catch(...)
+            {
+                ctx.local.setError("unknown exeption");;
```
Comment (EDDragonWolf) [inline include/thread_pool.h]: Extra `;`?

File: src/graft_manager.cpp [hunk 4]
```diff
@@ -208,44 +209,98 @@ void ClientRequest::createJob(Manager &manager)
...
+            }
+        }
+        catch(const std::exception& e)
+        {
+            setError(e.what());
+            m_prms.input.reset();
+        }
+        catch(...)
+        {
+            setError("unknown exeption");;
```
Comment (EDDragonWolf) [inline src/graft_manager.cpp]: same

File: src/graft_manager.cpp [hunk 5]
```diff
@@ -208,44 +209,98 @@ void ClientRequest::createJob(Manager &manager)
...
```
Comment (EDDragonWolf) [inline src/graft_manager.cpp]: also

**Pull request merged**